src/nodes.coffee

```diff
@@ -1399,7 +1421,7 @@ exports.Class = class Class extends Base
       method.name = new (if methodName.shouldCache() then Index else Access) methodName
       method.name.updateLocationDataIfMissing methodName.locationData
       method.ctor = (if @parent then 'derived' else 'base') if methodName.value is 'constructor'
-      method.error 'Methods cannot be bound functions' if method.bound
+      method.error 'Cannot define a constructor as a bound function' if method.bound and method.ctor
```

Comment:
In error messages, let’s use the phrasing “bound (fat arrow) function”. [MDN calls `=>` functions “arrow functions,”](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) so that’s what most people will probably know them as. I think we should be explicit about “fat” arrow to distinguish from `->`.